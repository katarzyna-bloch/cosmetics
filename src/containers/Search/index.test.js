import { mount } from 'enzyme'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { Search } from '.'

jest.mock('../../components/ProductsList', () => {
  const ProductsList = () => null
  return ProductsList
})

it('should render correctly', () => {
  const props = {
    fetchProducts: jest.fn(),
    searchProducts: [
      { 
        name: 'Test', 
        img: 'img.jpg', 
        id: '1', 
        price: '10z≈Ç',
      },
    ],
    location: {
      search: '?q=krem',
    },
    loading: false,
  }
  const wrapped = mount(<Search {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
  wrapped.update()
  expect(props.fetchProducts).toHaveBeenCalled()
})

it('should render correctly without products', () => {
  const props = {
    fetchProducts: jest.fn(),
    searchProducts: [],
    location: {
      search: '?q=krem',
    },
    loading: false,
  }
  const wrapped = shallow(<Search {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
