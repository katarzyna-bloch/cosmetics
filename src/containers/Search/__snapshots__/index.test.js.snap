// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Search
  fetchProducts={
    [MockFunction] {
      "calls": Array [
        Array [
          "?q=krem",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  location={
    Object {
      "search": "?q=krem",
    }
  }
  products={
    Array [
      Object {
        "id": "1",
        "img": "img.jpg",
        "name": "Test",
        "price": "10zł",
      },
    ]
  }
>
  <ProductsList
    products={
      Array [
        Object {
          "id": "1",
          "img": "img.jpg",
          "name": "Test",
          "price": "10zł",
        },
      ]
    }
  />
</Search>
`;
